colossalai/inference/core/engine.py

```diff
@@ -132,26 +132,39 @@ def _shardformer(
 
     def generate(
         self,
+        prompts: List[str] = None,
+        prompts_token_ids: Union[List[int], torch.Tensor, np.ndarray] = None,
         generation_config: GenerationConfig = None,
     ) -> List[str]:
         """
         Executing the inference step.
 
         Args:
+            prompts (Union[List[str], optional): Input prompts. Defaults to None.
+            prompts_token_ids (List[List[int]], optional): token ids of input prompts. Defaults to None.
             generation_config (GenerationConfig, optional): Huggingface GenerationConfig used for inference. Defaults to None.
 
         Returns:
             List[str]: Inference result returned by one generation.
         """
 
         self.generation_config = generation_config
+        self.add_request(prompts=prompts, prompts_token_ids=prompts_token_ids)
```

Comment:
only add request when prompts or prompt_token_ids is not None.

Reply:
If we receive a prompt or prompt token ids, we should also assert that there is no sequences left in the list.